src/aliceVision/image/dcp.cpp

```diff
@@ -0,0 +1,1425 @@
+            if (value[i + ofs] < 32 || value[i + ofs] > 126) {
+                isstring = false;
+            }
+
+        if (isstring) {
+            if (size < 3) {
+                return;
+            }
+
+            std::size_t j = 0;
+
+            for (i = 0; i + ofs < datasize && i < 64 && value[i + ofs]; i++) {
+                if (value[i + ofs] == '<' || value[i + ofs] == '>') {
+                    buffer[j++] = '\\';
+                    if (j > size - 2) {
+                        break;
+                    }
+                }
+
+                buffer[j++] = value[i + ofs];
+                if (j > size - 2) {
+                    break;
+                }
+            }
+
+            buffer[j++] = 0;
+            return;
+        }
+    }
+    else if (type == TagType::T_ASCII) {
+        snprintf(buffer, size, "%.64s", value + ofs);
+        return;
+    }
+
+    size_t maxcount = std::min<size_t>(datasize, 10);
+
+    buffer[0] = 0;
+
+    for (int i = 0; i < std::min<int>(maxcount, datasize * getTypeSize(type) - ofs); i++) {
+        std::size_t len = strlen(buffer);
+
+        if (i > 0 && size - len > 2) {
+            strcat(buffer, ", ");
+            len += 2;
+        }
+
+        char* b = buffer + len;
+
+        switch (type) {
+        case TagType::T_UNDEFINED:
+        case TagType::T_BYTE:
+            snprintf(b, size - len, "%d", value[i + ofs]);
+            break;
+
+        case TagType::T_SSHORT:
+            snprintf(b, size - len, "%d", toInt(2 * i + ofs));
+            break;
+
+        case TagType::T_SHORT:
+            snprintf(b, size - len, "%u", toInt(2 * i + ofs));
+            break;
+
+        case TagType::T_SLONG:
+            snprintf(b, size - len, "%d", toInt(4 * i + ofs));
+            break;
+
+        case TagType::T_LONG:
+            snprintf(b, size - len, "%u", toInt(4 * i + ofs));
+            break;
+
+        case TagType::T_SRATIONAL:
+            snprintf(b, size - len, "%d/%d", (int)sget4(value + 8 * i + ofs, order), (int)sget4(value + 8 * i + ofs + 4, order));
+            break;
+
+        case TagType::T_RATIONAL:
+            snprintf(b, size - len, "%u/%u", (uint32_t)sget4(value + 8 * i + ofs, order), (uint32_t)sget4(value + 8 * i + ofs + 4, order));
+            break;
+
+        case TagType::T_FLOAT:
+            snprintf(b, size - len, "%g", toDouble(8 * i + ofs));
+            break;
+
+        default:
+            break;
+        }
+    }
+
+    if (datasize > maxcount && size - strlen(buffer) > 3) {
+        strcat(buffer, "...");
+    }
+}
+
+std::string Tag::valueToString() const
+{
+    char buffer[1024];
+    toString(buffer, sizeof(buffer));
+    return buffer;
+}
+
+Tag* findTag(TagKey tagID, std::vector<Tag>& v_Tags)
+{
+    size_t idx = 0;
+    while (idx < v_Tags.size() && (v_Tags[idx].tagID != (unsigned short)tagID))
+    {
+        idx++;
+    }
+    return (idx == v_Tags.size()) ? nullptr : &v_Tags[idx];
```

Comment:
use `find()`, something like this 
```suggestion
    const auto res = find(v_Tags.begin(), v_Tags.end(), tagID);
    return res == v_Tags.end() ? nullptr : res;
```